test/unit_tests/braket/experimental/autoqasm/test_api.py

```diff
@@ -832,7 +884,7 @@ def prog() -> None:
 __bit_0__ = measure __qubits__[0];
 a = __bit_0__;
 b = a;"""
-    assert prog().to_ir() == expected
+    assert prog.to_ir() == expected
 
 
 def test_nested_function():
```

Comment:
`test_nested_function` fails now because the inner function `ghz` is a pure Python function but is now being called with a `FreeParameter`, which causes infinite recursion. (When called with an integer `n >= 1`, it terminates at some point.) Not sure we can really avoid this.

I guess a simple way to fix this would be to use a Python variable instead of a FreeParameter, for example:

```
def test_nested_function():
    ghz_size = 5

    @aq.main
    def make_ghz() -> None:
        def ghz(n: int):
            if n == 1:
                h(0)
            else:
                ghz(n - 1)
                cnot(0, n - 1)

        ghz(ghz_size)

    expected = """OPENQASM 3.0;
qubit[5] __qubits__;
h __qubits__[0];
cnot __qubits__[0], __qubits__[1];
cnot __qubits__[0], __qubits__[2];
cnot __qubits__[0], __qubits__[3];
cnot __qubits__[0], __qubits__[4];"""

    assert make_ghz.to_ir() == expected
```

I tested this and it seems to be fine.

Reply:
This is awesome! Definitely would love to see us handle free parameters as function inputs for recursive functions. I wonder if there's some check we can do while converting a function to turn off recursive mode if parameters are not initialized (or if that makes sense to do). At the very least we could catch this and raise a more explicit error